### ISSUE TITLE
EC2 MachineImage: support for cross account ssm parameter

### ISSUE THREAD

**Issue by danilodesolesonar** [aws/aws-cdk#31554]
### Describe the feature

`MachineImage.from_ssm_parameter(parameter_name=....)` doesn't support cross-account parameters. You cannot provide an ARN. I believe the module should support fetching of AMI ID from shared SSM parameters.

### Use Case

It's very likely to publish SSM parameters with AMI ID from a centralised account, when creating new EC2 instances `MachineImage.from_ssm_parameter(parameter_name=....)` should support the lookup of them.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.159.1 (build c66f4e3)

### Environment details (OS name and version, etc.)

Linux

**Comment by pahud** [comment]
Yes it makes sense. aws-ssm actually supports [cross-account sharing](https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk-lib/aws-ssm#cross-account-ssm-parameters-sharing) but I am not sure if MachineImage allows you to specify that. We need more experiment with that so any further inputs here is welcome and appreciated.

Making this a p2, please help us prioritize with 👍 .

**Comment by ozeebee** [comment]
+1

**Comment by d1rrected** [comment]
+1

**Comment by oleg-boulanov** [comment]
+100 !

**Comment by oleg-boulanov** [comment]
The issue is not only with CDK, but with CloudFormation itself also.
Can you please raise the issue priority?
Shared parameters need to be supported throughout all systems, otherwise they don't make much sense.
